sdk/core/Azure.Core/src/Pipeline/RetryPolicy.cs

```diff
@@ -264,7 +264,7 @@ private TimeSpan GetNextDelayInternal(HttpMessage message)
         {
             return _delayStrategy.GetNextDelay(
                 message.HasResponse ? message.Response : default,
-                message.RetryNumber);
+                message.RetryNumber + 1);
```

Comment:
This is because we're essentially asking, "what is the delay for the next retry that we're going to do?"

Reply:
Exactly - the retry number on the message gets incremented after we do the delay.